### PR TITLE
[PROD] embed payment in website

### PR DESCRIPTION
**Pull request by antoineeudes**

### CODE REVIEW

File: functions/payment.js [hunk 1]
```diff
@@ -18,66 +17,30 @@ exports.getSessionId = functions.https.onCall(async (data, context) => {
...
-        },
-        quantity: 1,
-      },
-    ],
-    mode: "payment",
-    success_url: `${data.returnUrl}?session_id={CHECKOUT_SESSION_ID}&redirectTo=payment_success`,
-    cancel_url: data.returnUrl,
+  const paymentIntent = await stripe.paymentIntents.create({
+    amount: data.price,
+    currency: "gbp",
```
Comment (BenEllerby) [inline functions/payment.js:31]: will it only be GBP

File: src/components/organisms/PaymentModal/CardSection.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const CARD_ELEMENT_OPTIONS = {
+  style: {
+    base: {
+      color: "white",
+      fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
+      fontSmoothing: "antialiased",
+      fontSize: "16px",
+      "::placeholder": {
+        color: "#aab7c4",
```
Comment (BenEllerby) [inline src/components/organisms/PaymentModal/CardSection.tsx:13]: colors file

File: src/components/organisms/PaymentModal/CardSection.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { CardElement } from "@stripe/react-stripe-js";
+import "./CardSectionStyles.scss";
+
+const CARD_ELEMENT_OPTIONS = {
+  style: {
+    base: {
+      color: "white",
+      fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
```
Comment (BenEllerby) [inline src/components/organisms/PaymentModal/CardSection.tsx:9]: not standard for proj?

File: src/components/organisms/PaymentModal/CardSection.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      fontSmoothing: "antialiased",
+      fontSize: "16px",
+      "::placeholder": {
+        color: "#aab7c4",
+      },
+    },
+    invalid: {
+      color: "#fa755a",
```
Comment (BenEllerby) [inline src/components/organisms/PaymentModal/CardSection.tsx:17]: ''

File: src/components/organisms/PaymentModal/CardSectionStyles.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
+@import "scss/constants.scss";
+
+.StripeElement {
+  height: 40px;
+  padding: 10px;
+  width: 100%;
+  color: #32325d;
```
Comment (BenEllerby) [inline src/components/organisms/PaymentModal/CardSectionStyles.scss:7]: color file

File: src/components/organisms/PaymentModal/CardSectionStyles.scss [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  transition: box-shadow 150ms ease;
+  border-radius: 22px;
+}
+
+.StripeElement--focus {
+  border: 1px solid $light-grey;
+}
+
+.StripeElement--invalid {
+  border-color: #fa755a;
```
Comment (BenEllerby) [inline src/components/organisms/PaymentModal/CardSectionStyles.scss:24]: color file

File: src/components/organisms/PaymentModal/PaymentForm.tsx [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      setErrorMessage(result.error.message);
+    } else {
+      if (result.paymentIntent?.status === "succeeded") {
+        setIsPaymentSuccess(true);
+      }
+    }
+    setIsFormBeingSubmitted(false);
+  };
+
+  if (!clientSecret) {
```
Comment (BenEllerby) [inline src/components/organisms/PaymentModal/PaymentForm.tsx:83]: ...

Comment (BenEllerby) [reply src/components/organisms/PaymentModal/PaymentForm.tsx:83]: Don't love this, you don't have a loading component that could listen to a flag prop

File: src/components/organisms/PaymentModal/PaymentModal.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  onHide: () => void;
+  selectedEvent: VenueEvent;
+}
+
+const PaymentModal: React.FunctionComponent<PropsType> = ({
+  show,
+  onHide,
+  selectedEvent,
+}) => {
+  useConnectUserPurchaseHistory();
```
Comment (BenEllerby) [inline src/components/organisms/PaymentModal/PaymentModal.tsx:22]: can you break this up into smaller atomic pieces

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx [hunk 9]
```diff
@@ -216,6 +204,13 @@ const JazzbarEntranceExperience: React.FunctionComponent<PropsType> = ({
             </div>
           </div>
         </div>
+        {user && selectedEvent && (
+          <PaymentModal
+            selectedEvent={selectedEvent}
+            show={isPaymentModalOpen}
+            onHide={() => setIsPaymentModalOpen(false)}
```
Comment (BenEllerby) [inline src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx:211]: anonymouse functions are not good for perf when in the render method.

File: functions/payment.js (hunk 1 above)
Comment (antoineeudes) [reply functions/payment.js:31]: for now it is, I'm going to ask the team this evening if they consider having multiple currencies

**Pull request merged**